src/main/java/io/rtr/conduit/amqp/impl/AMQPConsumerBuilder.java

```diff
@@ -165,16 +201,39 @@ protected boolean isPoisonQueueEnabled() {
     @Override
     protected void validate() {
         assertNotNull(exchange, "exchange");
+        if (dynamicQueueCreation && ensureBasicConfig) {
+            throw new IllegalArgumentException("Both dynamicQueueCreation and ensureBasicConfig are enabled.");
+        }
         if(!dynamicQueueCreation){
             assertNotNull(queue, "queue");
         }
         else{
             assertNotNull(dynamicQueueRoutingKey, "dynamicQueueRoutingKey");
         }
+        if (ensureBasicConfig) {
+            assertNotNull(queue, "queue");
+            assertNotNull(exchangeType, "exchangeType");
+            assertNotNull(routingKey, "routingKey");
```

Comment:
I'll switch it up so its automatically handled, so null == "" for the routing key.